src/wildcard-service/init.lua

```diff
@@ -42,40 +51,55 @@ function _M:get_apicast_servers()
     return ngx.exit(404)
   end
 
+  local resolver = resty_resolver:instance()
   local enviroments = {'production', 'sandbox'}
   for i=1, #enviroments do
     local env = enviroments[i]
     local api_url = self.api_host .. '/admin/api/services/proxy/configs/' .. env .. '.json?' .. query
     local response = self.http_client.get(api_url)
-
     if response.status == 200 then
-      configs = cjson.decode(response.body)
+      local configs = cjson.decode(response.body)
       if #configs.proxy_configs > 0 then
-        local servers = {}
         for z = 1, #configs.proxy_configs do
-          local proxy_config = configs.proxy_configs[z]
-          local proxy = cjson.decode(proxy_config.content).proxy
-          if env == 'production' then
-            servers[#servers + 1] = proxy.endpoint
+          local proxy_config = configs.proxy_configs[z].proxy_config
+          local proxy = proxy_config.content.proxy
+          local endpoint
+            
+          if env == 'production' then 
+            local url = resty_url.split(proxy.endpoint)
+            local _, _, _, host, _ = unpack(url)
+            if arg_host == host then
+              local resolver_servers = resolver:get_servers(self.apicast_production_endopoint, { port = self.apicast_port })
+              ngx.log(ngx.WARN, 'servers for enviroment: ', env, ' found: ', resolver_servers)
+              return resolver_servers
+            end
           else
-            servers[#servers + 1] = proxy.sandbox_endpoint
+            local url = resty_url.split(proxy.sandbox_endpoint)
+            local _, _, _, host, _ = unpack(url)
+            if arg_host == host then
+              local resolver_servers = resolver:get_servers(self.apicast_staging_endpoint, { port = self.apicast_port })
+              ngx.log(ngx.WARN, 'servers for enviroment: ', env, ' found: ', resolver_servers)
```

Comment:
WARN ? Shouldn't that be some INFO level? It happens on every requests, that should not be a warning.